server/schedule/replica_checker.go

```diff
@@ -201,8 +201,8 @@ func (r *ReplicaChecker) checkBestReplacement(region *core.RegionInfo) *Operator
 		checkerCounter.WithLabelValues("replica_checker", "all_right")
 		return nil
 	}
-	storeID, newScore := r.selectBestReplacement(region, oldPeer)
-	if storeID == 0 {
+	newPeer, newScore := r.SelectBestReplacedPeerToAddReplica(region, oldPeer)
```

Comment:
I was intentional to return `storeID` instead of `peer` to avoid alloc too many IDs during replacement check.